### PR TITLE
[Datastore] Fix reading parquet directory when using datastore profiles

### PR DESCRIPTION
**Pull request by alxtkr77**
https://jira.iguazeng.com/browse/ML-4751
When a datastore profile directory (as opposed to a single file) is utilized to read a pandas dataframe, the appropriate filesystem is already determined and set up. However, it remains unused. This is because, while pandas attempts to create a new instance of the filesystem, mlrun doesn't forward the filesystem to pandas. As a result, pandas encounters an error due to its unfamiliarity with the 'ds://' schema. This PR rectifies this by providing the pre-established filesystem to the pandas parquet reader and removing the 'ds://prfofile' prefix from the directory URL

### CODE REVIEW

File: tests/integration/aws_s3/test_aws_s3.py [hunk 1]
```diff
@@ -198,3 +202,35 @@ def test_using_profile(self, use_datastore_profile):
...
+        data1 = {"Column1": [1, 2, 3], "Column2": ["A", "B", "C"]}
+
+        # Define data for the second DataFrame
+        data2 = {"Column1": [4, 5, 6], "Column2": ["X", "Y", "Z"]}
+
+        # Create the DataFrames
+        df1 = pd.DataFrame(data1)
+        df2 = pd.DataFrame(data2)
+        temp_file1 = tempfile.mktemp(suffix=".parquet")
+        temp_file2 = tempfile.mktemp(suffix=".parquet")
```
Comment (tomerm-iguazio) [inline tests/integration/aws_s3/test_aws_s3.py:221]: Please use with statement, in order to delete the file after the test.

Comment (tomerm-iguazio) [reply tests/integration/aws_s3/test_aws_s3.py:221]: with tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file1, \
                tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file2:

Comment (tomerm-iguazio) [reply tests/integration/aws_s3/test_aws_s3.py:221]: with tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file1, \
                tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file2:

Comment (alxtkr77) [reply tests/integration/aws_s3/test_aws_s3.py:223]: Why to use  tempfile.NamedTemporaryFile(...) ? I need only name, not opening the file

File: tests/integration/aws_s3/test_aws_s3.py [hunk 2]
```diff
@@ -198,3 +202,35 @@ def test_using_profile(self, use_datastore_profile):
         # cleanup
         for param in params:
             os.environ.pop(param)
+
+    def test_directory(self, use_datastore_profile):
+        param = self.s3["ds"] if use_datastore_profile else self.s3["s3"]
```
[the remark this answers is not in the document]
Comment (alxtkr77) [reply tests/integration/aws_s3/test_aws_s3.py:207]: Fixed

File: mlrun/datastore/base.py [hunk 3]
```diff
@@ -285,7 +285,14 @@ def reader(*args, **kwargs):
         if file_system:
             if self.supports_isdir() and file_system.isdir(file_url) or df_module == dd:
                 storage_options = self.get_storage_options()
-                if storage_options:
+                if url.startswith("ds://"):
```
[the remark this answers is not in the document]
Comment (assaf758) [reply mlrun/datastore/base.py:288]: Good observation Tomer, however not that the data-store profile mechanism is intended to be used as part of mlrun abstractions (source, target, data-item etc) - and not to be used as regular new "file-system" with direct pandas API.
The later would require another approach entirely as data-store profiles assume access to mlrun secrets etc.

File: tests/integration/aws_s3/test_aws_s3.py (hunk 1 above)
Comment (tomerm-iguazio) [reply tests/integration/aws_s3/test_aws_s3.py:223]: It is better to delete the file after the test.

**Review by tomerm-iguazio — APPROVED** [review]
looking good!

**Pull request merged**